packages/message/src/DeadLetterMessage.ts

```diff
@@ -0,0 +1,54 @@
+import { SQSRecord, EventBridgeEvent } from 'aws-lambda';
+
+import { parseSQSMessageBody } from '@cumulus/aws-client/SQS';
+import { CumulusMessage } from '@cumulus/types/message';
+import Logger from '@cumulus/logger';
+
+import { getCumulusMessageFromExecutionEvent } from './StepFunctions';
+
+const log = new Logger({ sender: '@cumulus/DeadLetterMessage' });
+
+type StepFunctionEventBridgeEvent = EventBridgeEvent<'Step Functions Execution Status Change', { [key: string]: string }>;
+type UnwrapDeadLetterCumulusMessageReturnType = (
+  StepFunctionEventBridgeEvent
+  | AWS.SQS.Message
+  | SQSRecord
+  | CumulusMessage
+);
+
+/**
+ * Unwrap dead letter Cumulus message, which may be wrapped in a
+ * States cloudwatch event, which is wrapped in an SQS message.
+ *
+ * @param {Object} messageBody - received SQS message
+ * @returns {Object} the cumulus message or nearest available object
+ */
+export const unwrapDeadLetterCumulusMessage = async (
+  messageBody: UnwrapDeadLetterCumulusMessageReturnType
```

Comment:
Is it possible/preferable to refactor by making this function only receive non-Cumulus messages as input and then a separate function that only even returns a Cumulus message? In general my thinking is that it might be good to have the type safety of a function that only returns a Cumulus message given diverse inputs. But perhaps that's not feasible.

Reply:
Enitrely fair, I was erring on the side of conservative refactoring here given the scope of the other refactors and this being an operational issue.      I'll time-box give a refactor a shot here.

Reply:
In thinking about this, I think the best approach short of unwinding the recursion entirely in favor of branch logic/throwing if an unmet case occurs is probably to do a type eval in the calling method.    That said, both of the callers are javascript currently (thanks API package), and the in the case of the function in question here, will throw on an un-processable message object at some point.....

That said, this `  if (!isPostRDSDeploymentExecution(cumulusMessage)) {` check happening without validation almost certainly doesn't do what we want, and we should just throw up front.     I'm not sure I want to re-build a message validator outside of TS (we have typeguards in the CMA for example), but a top level key eval might not be uncalled for.    🤔

Reply:
I added an explicit catch/throw in the .js, but I low-key hate it.   Posting for discussion: b0c39d9aa

Reply:
Yeah I would ditch the try/catch. My concerns were more around making this function more deterministic in terms of what input it accepts and output it produces, probably by splitting it up into multiple functions. If that seems infeasible then so be it